gcp/apis/compute/v1alpha1/subnetwork_types.go

```diff
@@ -0,0 +1,298 @@
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   SubnetworkSpec   `json:"spec,omitempty"`
+	Status SubnetworkStatus `json:"status,omitempty"`
+}
+
+// SetBindingPhase of this Subnetwork.
+func (s *Subnetwork) SetBindingPhase(p v1alpha1.BindingPhase) {
+	s.Status.SetBindingPhase(p)
+}
+
+// SetConditions of this Subnetwork.
+func (s *Subnetwork) SetConditions(c ...v1alpha1.Condition) {
+	s.Status.SetConditions(c...)
+}
+
+// GetBindingPhase of this Subnetwork.
+func (s *Subnetwork) GetBindingPhase() v1alpha1.BindingPhase {
+	return s.Status.GetBindingPhase()
+}
+
+// SetClaimReference of this Subnetwork.
+func (s *Subnetwork) SetClaimReference(r *corev1.ObjectReference) {
+	s.Spec.ClaimReference = r
+}
+
+// GetClaimReference of this Subnetwork.
+func (s *Subnetwork) GetClaimReference() *corev1.ObjectReference {
+	return s.Spec.ClaimReference
+}
+
+// SetClassReference of this Subnetwork.
+func (s *Subnetwork) SetClassReference(r *corev1.ObjectReference) {
+	s.Spec.ClassReference = r
+}
+
+// GetClassReference of this Subnetwork.
+func (s *Subnetwork) GetClassReference() *corev1.ObjectReference {
+	return s.Spec.ClassReference
+}
+
+// SetWriteConnectionSecretToReference of this Subnetwork.
+func (s *Subnetwork) SetWriteConnectionSecretToReference(r corev1.LocalObjectReference) {
+	s.Spec.WriteConnectionSecretToReference = r
+}
+
+// GetWriteConnectionSecretToReference of this Subnetwork.
+func (s *Subnetwork) GetWriteConnectionSecretToReference() corev1.LocalObjectReference {
+	return s.Spec.WriteConnectionSecretToReference
+}
+
+// GetReclaimPolicy of this Subnetwork.
+func (s *Subnetwork) GetReclaimPolicy() v1alpha1.ReclaimPolicy {
+	return s.Spec.ReclaimPolicy
+}
+
+// SetReclaimPolicy of this Subnetwork.
+func (s *Subnetwork) SetReclaimPolicy(p v1alpha1.ReclaimPolicy) {
+	s.Spec.ReclaimPolicy = p
+}
+
+// GCPSubnetworkSpec contains fields of googlecompute.Subnetwork object that are
+// configurable by the user, i.e. the ones that are not marked as [Output Only]
+// In the future, this can be generated automatically.
+type GCPSubnetworkSpec struct {
+	// Description: An optional description of this resource. Provide this
+	// property when you create the resource. This field can be set only at
+	// resource creation time.
+	Description string `json:"description,omitempty"`
+
+	// EnableFlowLogs: Whether to enable flow logging for this subnetwork.
+	// If this field is not explicitly set, it will not appear in get
+	// listings. If not set the default behavior is to disable flow logging.
+	EnableFlowLogs bool `json:"enableFlowLogs,omitempty"`
+
+	// IPCidrRange: The range of internal addresses that are owned by this
+	// subnetwork. Provide this property when you create the subnetwork. For
+	// example, 10.0.0.0/8 or 192.168.0.0/16. Ranges must be unique and
+	// non-overlapping within a network. Only IPv4 is supported. This field
+	// can be set only at resource creation time.
+	IPCidrRange string `json:"ipCidrRange"`
```

Comment:
It's `IpCidrRange`, `PrivateIpGoogleAcces` and `id` in their API actually https://github.com/crossplaneio/crossplane/pull/699/files#diff-0f42cfd77ba976e7645754d011066667R46

Fixed. This will be something we'd need to discuss how to handle in general if we go with code-generation.

Reply:
Would it be worth opening an issue to explore whether we want to strive for consistency vs mirroring of provider API? Maybe this is just part of the overall code generation effort.